### PR TITLE
Add pre-commit formatting hooks

### PR DESCRIPTION
**Pull request by michplunkett**
Issue: https://github.com/Matico-Platform/matico/issues/211

## Overview

Added pre-commit hooks and removed `pinst` as the application is marked as `"private": true,` in the `package.json`.

### General Notes

Once this is merged, we can close https://github.com/Matico-Platform/matico/issues/211.

### Testing
- [x] Tested commits with Github GUI
- [x] Tested commits with Github CLI

<details><summary>Testing CLI commit</summary>

```
michaelp@MacBook-Air-3 matico % git commit -m "Testing pre-commit hook"
matico_admin/components/ColumnDetails.tsx 174ms
matico_admin/components/DatasetCreation/CSVFilePreviewer.tsx 40ms
matico_admin/components/DatasetCreation/FilePreviewerInterface.tsx 7ms
matico_admin/components/DatasetCreation/GeoJSONFilePreviewer.tsx 21ms
matico_admin/components/DatasetCreation/NewDatasetModal.tsx 7ms
matico_admin/components/DatasetCreation/NewUploadDatasetForm.tsx 15ms
matico_admin/components/DatasetCreation/ShpFilePreviwer.tsx 16ms
matico_admin/components/DatasetCreation/Uploader.tsx 9ms
matico_admin/components/DatasetCreation/utils/getCSVPreview.ts 3ms
matico_admin/components/DatasetCreation/utils/getJsonPreview.ts 2ms
matico_admin/components/DatasetCreation/utils/getShpPreview.ts 4ms
matico_admin/components/DataTable.tsx 14ms
matico_admin/components/FeatureEditor.tsx 10ms
matico_admin/components/Layout.tsx 6ms
matico_admin/components/LoginSignUpDialog.tsx 10ms
matico_admin/components/MapView.tsx 20ms
matico_admin/components/Nav.tsx 4ms
matico_admin/components/NewApiDialog.tsx 4ms
...
matico_types/spec/Variable.ts 0ms
matico_types/spec/VarOr.ts 1ms
matico_types/spec/View.ts 1ms
matico_types/spec/WASMCompute.ts 1ms
[add_pre_commit_hook 5e4baa0c] Testing pre-commit hook
 1 file changed, 1 insertion(+), 1 deletion(-)
michaelp@MacBook-Air-3 matico % 
```
</details>

- [x]  Did a `yarn build` and it looks fine

<details><summary>yarn build output</summary>

```
michaelp@MacBook-Air-3 matico % yarn build       
▲ [WARNING] Cannot find base config file "./node_modules/gts/tsconfig-google.json" [tsconfig.json]

    ../tsconfig.json:2:13:
      2 │   "extends": "./node_modules/gts/tsconfig-google.json",
        ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

▲ [WARNING] Cannot find base config file "./node_modules/gts/tsconfig-google.json" [tsconfig.json]

    ../tsconfig.json:2:13:
      2 │   "extends": "./node_modules/gts/tsconfig-google.json",
        ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

1 warning

  dist/index.js      142.3kb
  dist/index.js.map  292.5kb

⚡ Done in 57ms
1 warning

  dist/bundle.js      250.9kb
  dist/bundle.js.map    1.1mb

⚡ Done in 72ms
▲ [WARNING] Cannot find base config file "./node_modules/gts/tsconfig-google.json" [tsconfig.json]

    ../tsconfig.json:2:13:
      2 │   "extends": "./node_modules/gts/tsconfig-google.json",
        ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

▲ [WARNING] Cannot find base config file "./node_modules/gts/tsconfig-google.json" [tsconfig.json]

    ../tsconfig.json:2:13:
      2 │   "extends": "./node_modules/gts/tsconfig-google.json",
        ╵              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

1 warning

  dist/index.js  2.1mb ⚠️

⚡ Done in 202ms
Prisma schema loaded from schema.prisma

✔ Generated Prisma Client (4.2.1 | library) to ./../node_modules/@prisma/client in 52ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```
info  - Skipping validation of types

./pages/_app.tsx
27:13  Warning: Do not include stylesheets manually. See: https://nextjs.org/docs/messages/no-css-tags.  @next/next/no-css-tags

./pages/apps/demo.tsx
44:6  Warning: React Hook useEffect has missing dependencies: 'createAppFromDemo' and 'router'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
87:6  Warning: React Hook useMemo has a missing dependency: 'MaticoApp'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

./pages/apps/edit/[...params].tsx
111:6  Warning: React Hook useMemo has missing dependencies: 'MaticoApp', 'error', and 'onUpdateSpec'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./pages/index.tsx
176:6  Warning: React Hook useEffect has missing dependencies: 'session.user.email' and 'session.user.name'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps

./components/DatasetCreation/FilePreviewer.tsx
58:17  Warning: Missing "key" prop for element in iterator  react/jsx-key
62:28  Warning: Missing "key" prop for element in iterator  react/jsx-key

./components/DatasetCreation/Uploader.tsx
69:6  Warning: React Hook useEffect has missing dependencies: 'metadata', 'onDone', 'onFail', and 'table'. Either include them or remove the dependency array. If 'onDone' changes too often, find the parent component that defines it and wrap that definition in useCallback.  react-hooks/exhaustive-deps

./components/Header/Header.tsx
70:13  Warning: Do not use <img>. Use Image from 'next/image' instead. See https://nextjs.org/docs/messages/no-img-element.  @next/next/no-img-element

./components/NewAppModal/NewAppModal.tsx
45:13  Warning: Image elements must have an alt prop, either with meaningful text, or an empty string for decorative images.  jsx-a11y/alt-text

./components/TemplateSelector/TemplatesSelector.tsx
120:11  Warning: passHref is missing. See https://nextjs.org/docs/messages/link-passhref  @next/next/link-passhref
126:11  Warning: passHref is missing. See https://nextjs.org/docs/messages/link-passhref  @next/next/link-passhref
147:13  Warning: Missing "key" prop for element in iterator  react/jsx-key

./components/UserFinder/UserFinder.tsx
37:17  Warning: Do not use <img>. Use Image from 'next/image' instead. See https://nextjs.org/docs/messages/no-img-element.  @next/next/no-img-element
37:17  Warning: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images.  jsx-a11y/alt-text

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
info  - Linting  
info  - Creating an optimized production build  
warn  - Compiled with warnings

./node_modules/@maticoapp/matico_components/dist/index.js
Critical dependency: the request of a dependency is an expression

Import trace for requested module:
./pages/apps/demo.tsx

info  - Collecting page data  
info  - Generating static pages (5/5)
info  - Finalizing page optimization  

Route (pages)                                      Size     First Load JS
┌ λ /                                              2.53 kB         208 kB
├   /_app                                          0 B             135 kB
├ ○ /404                                           195 B           135 kB
├ λ /api/apps                                      0 B             135 kB
├ λ /api/apps/[id]                                 0 B             135 kB
├ λ /api/apps/[id]/collaborators                   0 B             135 kB
├ λ /api/apps/[id]/collaborators/[collaboratorId]  0 B             135 kB
├ λ /api/apps/[id]/permisions                      0 B             135 kB
├ λ /api/auth/[...nextauth]                        0 B             135 kB
├ λ /api/compute                                   0 B             135 kB
├ λ /api/compute/list                              0 B             135 kB
├ λ /api/datasets                                  0 B             135 kB
├ λ /api/datasets/[id]                             0 B             135 kB
├ λ /api/datasets/upload                           0 B             135 kB
├ λ /api/users                                     0 B             135 kB
├ λ /api/users/[userId]                            0 B             135 kB
├ λ /apps/[...params]                              4.11 kB         142 kB
├ ○ /apps/demo                                     1.69 kB         461 kB
├ λ /apps/edit/[...params]                         1.16 kB         440 kB
├ λ /datasets                                      1.56 kB         433 kB
├ ○ /privacy                                       3.5 kB          188 kB
├ ○ /tos                                           4.13 kB         189 kB
├ λ /users                                         713 B           192 kB
└ λ /users/[userId]/[appName]                      1.53 kB         137 kB
+ First Load JS shared by all                      145 kB
  ├ chunks/framework-c5b7bbbf6345f581.js           45.3 kB
  ├ chunks/main-89dcc7b60dfcb1eb.js                30.9 kB
  ├ chunks/pages/_app-6f8812894446b267.js          55.7 kB
  ├ chunks/webpack-5915b0363dccb3bc.js             3.36 kB
  └ css/dd3c1e3e386a927d.css                       9.93 kB

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)

michaelp@MacBook-Air-3 matico % 
```
</deatils>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,6 @@
     "**/webpack**",
     "**/babel-loader**"
   ],
-  "private": false,
```
Comment (michplunkett) [inline package.json:31]: Above this on line 9, we have `"private": true,` in the file so I removed `pinst`.

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+yarn prettier-write
+yarn rust-format
```
Comment (michplunkett) [inline .husky/pre-commit:5]: The two pre-commit commands.

File: matico_components/src/Utils/paneGrid.tsx [hunk 3]
```diff
@@ -3,16 +3,20 @@ import styled from "styled-components";
 export const PaneGrid = styled.div`
     display: grid;
     position: relative;
-    grid-template-columns: clamp(10px, 2.5rem, 15%) clamp(10px, 2.5rem, 15%) 1fr clamp(
+    grid-template-columns:
```
Comment (michplunkett) [inline matico_components/src/Utils/paneGrid.tsx:6]: The formatting doing its thing.

**Review by stuartlynn — APPROVED** [review]
This looks great. Merging now

**Pull request merged**